module/os/linux/zfs/zfs_vfsops.c

```diff
@@ -2275,6 +2278,7 @@ zfs_resume_fs(zfsvfs_t *zfsvfs, dsl_dataset_t *ds)
 	    zp = list_next(&zfsvfs->z_all_znodes, zp)) {
 		err2 = zfs_rezget(zp);
 		if (err2) {
+			spl_d_drop_aliases(ZTOI(zp));
 			remove_inode_hash(ZTOI(zp));
 			zp->z_is_stale = B_TRUE;
```

Comment:
If we can accomplish all of the invalidation here then `zp->z_is_stale` should be unused and can be removed.

Reply:
Will check.

Reply:
Yup, but I'm keeping the code for `zpl_revalidate` for now; is there any notion of devel-only configure time flags that could be passed to ZFS and translated as preprocessor defines?

That way, we could change the `#ifdef 0` to only remove `zpl_revalidate` for developers for now.

Reply:
Not really.  We have used module options to dynamically disable/enable certain functionality to make testing easier.  However, usually in the final version to be merged, after we're sure everything is working as intended, those module options are removed.  You could do something similar while iterating and testing this.